webanno-brat/src/main/java/de/tudarmstadt/ukp/clarin/webanno/brat/annotation/BratAnnotationEditor.java

```diff
@@ -179,7 +185,34 @@ public void onEndRequest(RequestCycle aCycle)
                         return;
                     }
                 }
-
+                
+				/*  */
+				// Whenever an action should be performed, do ONLY perform this action and nothing else, and only if the item actually is an action item
+                if("doAction".equals(action)){
+                	Project proj = aModel.getObject().getProject();
+                	StringValue layer_type = request.getParameterValue(PARAM_SPAN_TYPE);
+                	if(!layer_type.isEmpty()){
+	                	long layer_id = Long.parseLong(layer_type.beforeFirst('_'));
+	                	AnnotationLayer anno_layer = annotationService.getLayer(layer_id);
+	                	if(!StringUtils.isEmpty(anno_layer.getOnClickJavascriptAction())){ 
+		                	/* parse the action */
+	                		List<AnnotationFeature> anno_layer_features = annotationService.listAnnotationFeature(anno_layer);
+	                		AnnotationFS anno = WebAnnoCasUtil.selectByAddr(jCas, paramId.getId()); 
+		                	String js = onClickActionParser.parse(
+		                			anno_layer.getOnClickJavascriptAction(),
+		                			anno_layer,
+		                			anno_layer_features,
+		                			proj, 
+		                			aModel.getObject().getDocument(), 
+		                			anno);
+		                	aTarget.appendJavaScript(js);
```

Comment:
See comment in issue. I believe allowing for arbitrary JS may be going too complex.